File: smoke_tests/typescript/weak_hash.ts
// Weak Hash (MD5/SHA1) Test Cases

import * as crypto from 'crypto';

// Test 1: MD5 hash usage
function hashPasswordMD5(password: string): string {
    // VULNERABLE: MD5 is cryptographically broken
    return crypto.createHash('md5').update(password).digest('hex');
}

// Test 2: SHA1 hash usage
function hashPasswordSHA1(password: string): string {
    // VULNERABLE: SHA1 is cryptographically weak
    return crypto.createHash('sha1').update(password).digest('hex');
}

// Test 3: MD5 for file integrity
function calculateFileChecksumMD5(data: Buffer): string {
    // VULNERABLE: MD5 should not be used for security purposes
    const hash = crypto.createHash('md5');
    hash.update(data);
    return hash.digest('hex');
}

// Test 4: SHA1 for token generation
function generateTokenSHA1(userId: string): string {
    const timestamp = Date.now().toString();
    // VULNERABLE: SHA1 is not suitable for security tokens
    return crypto.createHash('sha1').update(userId + timestamp).digest('hex');
}

// Test 5: MD5 in HMAC (still weak)
function createHMACMD5(data: string, key: string): string {
    // VULNERABLE: Even in HMAC, MD5 is not recommended
    return crypto.createHmac('md5', key).update(data).digest('hex');
}

File: smoke_tests/rust/weak_hash.rs
// Weak Hash (MD5/SHA1) Test Cases

use md5;
use sha1::{Sha1, Digest};

// Test 1: MD5 for password hashing
fn hash_password_md5(password: &str) -> String {
    // VULNERABLE: MD5 is cryptographically broken
    format!("{:x}", md5::compute(password.as_bytes()))
}

// Test 2: SHA1 for password hashing
fn hash_password_sha1(password: &str) -> String {
    // VULNERABLE: SHA1 is cryptographically weak
    let mut hasher = Sha1::new();
    hasher.update(password.as_bytes());
    format!("{:x}", hasher.finalize())
}

// Test 3: MD5 for token generation
fn generate_token_md5(user_id: &str) -> String {
    use std::time::{SystemTime, UNIX_EPOCH};
    let timestamp = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
    let data = format!("{}{}", user_id, timestamp);
    // VULNERABLE: MD5 should not be used for security tokens
    format!("{:x}", md5::compute(data.as_bytes()))
}

// Test 4: SHA1 for file integrity checking
fn calculate_file_checksum_sha1(data: &[u8]) -> String {
    // VULNERABLE: SHA1 is not suitable for security purposes
    let mut hasher = Sha1::new();
    hasher.update(data);
    format!("{:x}", hasher.finalize())
}

// Test 5: MD5 digest for API signatures
fn create_api_signature_md5(payload: &str, secret: &str) -> String {
    let combined = format!("{}{}", payload, secret);
    // VULNERABLE: MD5 is not suitable for cryptographic signatures
    format!("{:x}", md5::compute(combined.as_bytes()))
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_weak_hashes() {
        let _ = hash_password_md5("password123");
        let _ = hash_password_sha1("secret");
    }
}
